skqulacs/circuit/pre_defined.py

```diff
@@ -79,6 +80,73 @@ def preprocess_x(x: List[float], index: int):
     return circuit
 
 
+xkeisuu = np.zeros([15, 15, 15])
+nCr = np.zeros([15, 15])
+
+
+def create_farhi_watle(
+    n_qubit: int, c_depth: int, seed: Optional[int] = None
+) -> LearningCircuit:
+    for i in range(15):
+        for j in range(i + 1):
+            nCr[i][j] = factorial(i) / factorial(j) / factorial(i - j)
+    for i in range(15):
+        for j in range(i):
+            if j == 0:
+                xkeisuu[i][0][i] = 1
+            else:
+                for k in range(i - j, i + 1):
+                    xkeisuu[i][j][k] = xkeisuu[i][j - 1][k] + nCr[i][j] * nCr[j][
+                        i - k
+                    ] * ((-1) ** (i + j + k))
+
+    def preprocess_x(x: List[float], index: int):
+        dex = index % len(x)
+        qubits_p_bit = ((n_qubit - dex) - 1) // len(x) + 1  # そのbitに割り当てられる量子の数
+        xa = (min(1, max(-1, x[dex])) + 1) / 2
+        sban = index // len(x)
+
+        xb = 0
+        if qubits_p_bit < 15:
+            for i in range(qubits_p_bit):
+                xb += xkeisuu[qubits_p_bit][sban][qubits_p_bit - i]
+                xb *= xa
+        else:
+            xb = xa  # あきらめた
```

Comment:
実装をあきらめたという意味の場合は、NotImplementedErrorを使って実装されてないということを明示するのが良いと思います。
```suggestion
            raise NotImplementedError("実装をあきらめた")
```

Reply:
nCrをあきらめてはいるけど、そのときはNotImplementedErrorじゃなくて通常版を返したい

Reply:
それはこのコードから読み取りづらいので、コメントとして追加しておきたいです。
```
            xb = xa # 実装が困難であるため、通常版を返す
```